internal/store/fleet.go

```diff
@@ -165,6 +168,9 @@ func (s *FleetStore) createOrUpdateTx(tx *gorm.DB, orgId uuid.UUID, resource *ap
 	fleet := model.NewFleetFromApiResource(resource)
 	fleet.OrgID = orgId
 
+	// Use the dedicated API to update annotations
+	fleet.Annotations = nil
```

Comment:
Should we log a warn if fleet. Annotations aren't before setting nil or at least document that it will be discarded on the method?

Reply:
I think logging a warning isn't warranted because it's fine to have annotations in the resource.  For example, if a user GETs a fleet, modifies it, and POSTs the yaml back.  Same for an internal task.  Anyway, we'll discuss it further because @mangelajo commented on the same thing in my other PR :)
But for now, I'd like to proceed instead of waiting to nail down the semantics of annotations.

https://github.com/flightctl/flightctl/pull/264#discussion_r1617001744

Reply:
OK to defer can we create an issue to track?